cpp/arcticdb/processing/aggregation.cpp

```diff
@@ -74,6 +61,87 @@ SegmentInMemory MinMaxAggregatorData::finalize(const std::vector<ColumnName>& ou
     return seg;
 }
 
+namespace
+{
+    void add_data_type_impl(DataType data_type, std::optional<DataType>& current_data_type) {
+        if (current_data_type.has_value()) {
+            auto common_type = has_valid_common_type(entity::TypeDescriptor(*current_data_type, 0),
+                                                     entity::TypeDescriptor(data_type, 0));
+            schema::check<ErrorCode::E_UNSUPPORTED_COLUMN_TYPE>(
+                    common_type.has_value(),
+                    "Cannot perform aggregation on column, incompatible types present: {} and {}",
+                    entity::TypeDescriptor(*current_data_type, 0), entity::TypeDescriptor(data_type, 0));
+            current_data_type = common_type->data_type();
+        } else {
+            current_data_type = data_type;
+        }
+    }
+
+    inline util::BitMagic::enumerator::value_type deref(util::BitMagic::enumerator iter) {
+        return *iter;
+    }
+
+    inline std::size_t deref(std::size_t index) {
+        return index;
+    }
+
+    template<typename T, typename T2=void>
+    struct OutputType;
+
+    template <typename InputType>
+    struct OutputType <InputType, typename std::enable_if_t<is_floating_point_type(InputType::DataTypeTag::data_type)>> {
+        using type = ScalarTagType<DataTypeTag<DataType::FLOAT64>>;
+    };
+
+    template <typename InputType>
+    struct OutputType <InputType, typename std::enable_if_t<is_unsigned_type(InputType::DataTypeTag::data_type)>> {
+        using type = ScalarTagType<DataTypeTag<DataType::UINT64>>;
+    };
+
+    template <typename InputType>
+    struct OutputType<InputType, typename std::enable_if_t<is_signed_type(InputType::DataTypeTag::data_type) && is_integer_type(InputType::DataTypeTag::data_type)>> {
+        using type = ScalarTagType<DataTypeTag<DataType::INT64>>;
+    };
+
+    template<>
+    struct OutputType<DataTypeTag<DataType::BOOL8>, void> {
+        using type = ScalarTagType<DataTypeTag<DataType::BOOL8>>;
+    };
+
+    template<>
+    struct OutputType<DataTypeTag<DataType::NANOSECONDS_UTC64>, void> {
+        using type = ScalarTagType<DataTypeTag<DataType::NANOSECONDS_UTC64>>;
+    };
+
+    template<>
+    struct OutputType<DataTypeTag<DataType::EMPTYVAL>, void> {
+        using type = ScalarTagType<DataTypeTag<DataType::EMPTYVAL>>;
+    };
+}
+
+/**********************
+ * AggregatorDataBase *
+ **********************/
+
+AggregatorDataBase::AggregatorDataBase(const AggregatorDataBase&)
```

Comment:
I guess this could be configured with clang-format. I prefer having opening curly-braces on their own lines since it makes the blocks easier to identify, but I that may be just because I'm used to it.